orderbook/src/fee.rs

```diff
@@ -134,19 +139,23 @@ where
             .await
     }
 
-    async fn is_valid_fee(&self, sell_token: H160, fee: U256) -> bool {
-        self.calculator.is_valid_fee(sell_token, fee).await
+    async fn is_valid_fee(&self, sell_token: H160, fee: U256, app_data: [u8; 32]) -> bool {
+        self.calculator
+            .is_valid_fee(sell_token, fee, app_data)
+            .await
     }
 }
 
 impl MinFeeCalculator {
+    #[allow(clippy::too_many_arguments)]
     fn new(
         price_estimator: Arc<dyn PriceEstimating>,
         gas_estimator: Arc<dyn GasPriceEstimating>,
         native_token: H160,
         measurements: Arc<dyn MinFeeStoring>,
         fee_factor: f64,
         bad_token_detector: Arc<dyn BadTokenDetecting>,
+        partner_additional_fee_factors: HashMap<H256, f64>,
```

Comment:
I'm confused that this is H256 while app_data is [u8; 32] but we convert one into the other. I feel like the type should be the same.

Reply:
I have chosen to do this because of the convenience for runtime arguments 

As an H256 app_id can be passed in as a hex string... as [u8;32] we would have to pass an array of bytes `[0, 0, 0, 0, 0... 1 ]`. If you have some suggestion on how we can parse the hex string directly into the correct format, I would be happy to apply it.

Reply:
You would do it with a custom parse function or wrapper type like in the comment below. We are already using https://docs.rs/hex/0.4.3/hex/ in some places which can turn a hex string into bytes.

Reply:
`H256` is so convenient :stuck_out_tongue:. That being said, I do agree with @vkgnosis that we are slightly abusing the type here.